packages/fx-core/src/plugins/resource/function/plugin.ts

```diff
@@ -340,9 +354,10 @@ export class FunctionPluginImpl {
     await this.syncConfigFromContext(ctx);
 
     if (
-      !this.config.functionAppName ||
-      !this.config.storageAccountName ||
-      !this.config.appServicePlanName
+      !isArmSupportEnabled() &&
```

Comment:
Just to confirm, the naming concatenation is done by arm, right?

Reply:
Synced offline and resolved question.